cli/src/ui.rs

```diff
@@ -92,7 +92,10 @@ impl UiOutput {
             scroll_past_eof: false,
             ..Default::default()
         };
-        let mut pager = streampager::Pager::new_using_stdio_with_config(streampager_config)?;
+        // Initialize with tty instead of stdio. Pager is allowed so long as
```

Comment:
(Optional) This comment and the commit description are a little confusing to me. Would you be able to clarify what's going on a bit?

Does tty replace stdin or stdout? I'm guessing it's the former, in which case I'd say something like:

> Initialize with tty instead of **stdin**. This will make the pager work whenever stdout is a tty.

But I'm not sure that's the correct interpretation.

Similarly, the commit message could be something like (assuming this interpretation is correct, the problem I'm trying to address is that I can't tell):

> Make our default pager work even if stdin is not a tty. This matches the behavior of `less`.
>
> `request_pager` doesn't check whether stdin is a tty, which was a bug before this commit.

Reply:
I'm also not sure that stdout has to be attached to a tty. The function that ends up being used, https://docs.rs/termwiz/latest/termwiz/terminal/unix/struct.UnixTerminal.html#tymethod.new, says "This will yield a terminal even if the stdio streams have been redirected, provided that the process has an associated controlling terminal.". Do we cause an error ourselves if stdout is not a tty? Should we? We might also want to mention this.

Reply:
> Does tty replace stdin or stdout?

Both. We won't spawn pager if `stdout != tty`, though. I'm not sure which part of my comment is unclear.

Reply:
Whether stdio means stdin, stdout, or both wasn't clear to me, especially given the second sentence, and the other context of this PR/bug. (I guess stdio also includes stderr, but we're probably not touching that). As you may notice, I couldn't guess correctly what you meant based on what you wrote.

So, I think clarifying that it's both seems valuable. 

Also, I'd mention that pager will fail if there's not tty at all, which seems be relevant to @lorentzforces .

Reply:
In case this helps follow my thinking: a major issue with the commit message is that the phrase "this seems good" in there could refer to the fact that "ui.request_pager() doesn't check if stdin is a tty". The fact that `less` is an example of a program that does something similar, as opposed to something we are using, is also not very clear.